systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesST.java

```diff
@@ -0,0 +1,93 @@
+ * https://github.com/strimzi/proposals/blob/main/022-feature-gates.md
+ */
+public class FeatureGatesST extends AbstractST {
+
+    static final String NAMESPACE = "feature-gates-tests";
+    private static final Logger LOGGER = LogManager.getLogger(FeatureGatesST.class);
+
+    private Deployment clusterOperator;
+    private List<EnvVar> defaultEnvVars;
+    private List<EnvVar> testEnvVars;
+
+    @BeforeAll
+    void setup(ExtensionContext extensionContext) {
+        installClusterWideClusterOperator(extensionContext, NAMESPACE, Constants.CO_OPERATION_TIMEOUT_DEFAULT, Constants.RECONCILIATION_INTERVAL);
+        clusterOperator = DeploymentResource.getDeploymentFromYaml(BundleResource.PATH_TO_CO_CONFIG);
+        defaultEnvVars = clusterOperator.getSpec().getTemplate().getSpec().getContainers().get(0).getEnv();
+    }
+
+    @AfterEach
+    void afterEach(ExtensionContext extensionContext) {
+        clusterOperator = redeployCOWithEnvVars(defaultEnvVars);
+        testEnvVars = null;
+    }
+
+    @BeforeEach
+    void beforeEachSetClusterOperator(ExtensionContext extensionContext) {
+        clusterOperator = DeploymentResource.getDeploymentFromYaml(BundleResource.PATH_TO_CO_CONFIG);
+        testEnvVars = clusterOperator.getSpec().getTemplate().getSpec().getContainers().get(0).getEnv();
+    }
+
+    protected Deployment redeployCOWithEnvVars(List<EnvVar> envVarList) {
+        // Apply env variables
+        clusterOperator.getSpec().getTemplate().getSpec().getContainers().get(0).setEnv(envVarList);
+        return new DeploymentBuilder(clusterOperator).build();
+    }
+
+    /**
+     * Control Plane Listener
+     * https://github.com/strimzi/proposals/blob/main/025-control-plain-listener.md
+     */
+    @Test
+    @ParallelNamespaceTest
+    public void testControlPlaneListenerFeatureGate(ExtensionContext extensionContext) {
+        final String namespaceName = extensionContext.getStore(ExtensionContext.Namespace.GLOBAL).get(Constants.NAMESPACE_KEY).toString();
+        final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+
+        testEnvVars.add(new EnvVar(Environment.STRIMZI_FEATURE_GATES_ENV, "+ControlPlaneListener", null));
+        redeployCOWithEnvVars(testEnvVars);
+        // Following line adds dependency on cluster-operator
+        resourceManager.createResource(extensionContext,
+                KafkaTemplates.kafkaPersistent(clusterName, 3).build());
+
+        LOGGER.info("Check for presence of ContainerPort 9090/tcp (tcp-ctrlplane) in first Kafka pod.");
+        final Pod kafkaPod = PodUtils.getPodsByPrefixInNameWithDynamicWait(namespaceName, clusterName + "-kafka-").get(0);
+        ContainerPort expectedControlPlaneContainerPort = new ContainerPort(9090, null, null, "tcp-ctrlplane", "TCP");
+        List<ContainerPort> kafkaPodPorts = kafkaPod.getSpec().getContainers().get(0).getPorts();
+        assertTrue(kafkaPodPorts.contains(expectedControlPlaneContainerPort));
```

Comment:
Now you are testing that CO is deployed with enabled ControlPlane listener, which is then present in the pod spec. But, this can be tested with some simple UT.

When it's the "listener", I would test if we are able to send messages or something like that.

Reply:
Yeah I will look on it.